frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -393,6 +418,10 @@ function simpleApiFactoryWithNamespace(
   }
 }
 
+function asQuery(queryParams?: QueryParameters): string {
+  return '?' + new URLSearchParams(queryParams).toString();
```

Comment:
For the requests to be exactly the same as before when the query is empty, the queryParams should be checked for validity (whether it's undefined or empty) before we return `'?'+..`, otherwise you may have `?` without any query.